clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientCallbackHandler.java

```diff
@@ -78,9 +82,15 @@ public void handle(Callback[] callbacks) throws UnsupportedCallbackException {
                 if (ac.isAuthorized())
                     ac.setAuthorizedID(authzId);
             } else if (callback instanceof ScramExtensionsCallback) {
-                ScramExtensionsCallback sc = (ScramExtensionsCallback) callback;
-                if (!SaslConfigs.GSSAPI_MECHANISM.equals(mechanism) && subject != null && !subject.getPublicCredentials(Map.class).isEmpty()) {
-                    sc.extensions((Map<String, String>) subject.getPublicCredentials(Map.class).iterator().next());
+                if (mechanism.startsWith("SCRAM") && subject != null && !subject.getPublicCredentials(Map.class).isEmpty()) {
```

Comment:
@rajinisivaram is this the correct way to check for the mechanism? I'm wondering why the previous code only checked for `!SaslConfigs.GSSAPI_MECHANISM.equals(mechanism)` and not other mechanisms as well